packages/flutter/lib/src/painting/alignment.dart

```diff
@@ -116,7 +115,7 @@ abstract class AlignmentGeometry {
   ///  * [Alignment], for which this is a no-op (returns itself).
   ///  * [AlignmentDirectional], which flips the horizontal direction
   ///    based on the `direction` argument.
-  Alignment resolve(TextDirection direction);
+  Alignment resolve(TextDirection? direction);
```

Comment:
Why do we allow `null` for direction?

Reply:
It looks like subclasses are asserting that this is non-null?

Reply:
It is nullable because null can be used via gradient.dart `Shader createShader(Rect rect, { TextDirection? textDirection })` or with ImageConfiguration that accepts null `textDirection`.